applications/nrf5340_audio/src/audio/streamctrl.c

```diff
@@ -222,11 +220,46 @@ void streamctrl_encoded_data_send(void const *const data, size_t len)
 	}
 }
 
+#if (CONFIG_AUDIO_TEST_TONE)
+#define TEST_TONE_BASE_FREQ_HZ 1000
+
+static int test_tone_button_press(void)
+{
+	int ret = 0;
+	static uint32_t test_tone_hz;
+
+	if (CONFIG_AUDIO_BIT_DEPTH_BITS != 16) {
+		LOG_WRN("Tone gen only supports 16 bits");
+		return -ECANCELED;
+	}
+
+	if (strm_state == STATE_STREAMING) {
+		if (test_tone_hz == 0) {
+			test_tone_hz = TEST_TONE_BASE_FREQ_HZ;
+		} else if (test_tone_hz >= TEST_TONE_BASE_FREQ_HZ * 4) {
+			test_tone_hz = 0;
+		} else {
+			test_tone_hz = test_tone_hz * 2;
+		}
+
+		if (test_tone_hz != 0) {
+			LOG_INF("Test tone set at %d Hz", test_tone_hz);
+		} else {
+			LOG_INF("Test tone off");
+		}
+
+		ret = audio_encode_test_tone_set(test_tone_hz);
+		ERR_CHK_MSG(ret, "Failed to generate test tone");
+	}
+
+	return ret;
```

Comment:
Will never return ret if ret is anything else than 0. So just return 0 and you don't have to set ret = 0 at the start.